src/basic/time-util.c

```diff
@@ -837,9 +839,14 @@ int parse_timestamp(const char *t, usec_t *usec) {
                 return -EINVAL;
 
         ret = (usec_t) x * USEC_PER_SEC + x_usec;
+        if (ret > USEC_TIMESTAMP_FORMATTABLE_MAX)
+                return -EINVAL;
```

Comment:
well, each guard one addition. If you permit both additions without individual checks then you might wrap around twice, hiding the overflow...
not that it matters really, but i wanted to be correct here ;-)